src/main.cpp

```diff
@@ -90,7 +90,12 @@ int main(int argc, char **argv){
 
     // Load heightmap
     std::vector<Door> doors;
-    geo::ShapePtr heightmap = createHeightMapShape(heightmap_filename, doors);
+    MapLoader loader;
+    cv::Mat mapImage;
+    nav_msgs::MapMetaData metadata;
+    loader.getMapImage(mapImage);
+    loader.getMapMetadata(metadata);
+    geo::ShapePtr heightmap = createHeightMapShape(mapImage, metadata.resolution, doors);
```

Comment:
This will segfault if no image was found above. for example if the user ctrl+Cs the program. Add a check to see it the image was indeed received. For example by checking loader.initialised. Take inspiration from the check below.